### PR TITLE
Vanilla Modal: scrolling content for keyboard users

### LINKED ISSUE #3196: Modal, Vanilla: AVT 2 issue: Keyboard users can't scroll static content when it overflows
## Environment

> Operating system

Mac

> Browser

All

> Automated testing tool and ruleset

n/a

> Assistive technology used to verify

Keyboard

## Detailed description

> What version of the Carbon Design System are you using?

v10.3.2

> What did you expect to happen?

I would be able to focus the content container so that I could use my arrow keys to scroll the content

> What happened instead?

I was not able to focus the container that had overflowing content so I was not able to read it

> What WCAG 2.1 checkpoint does the issue violate?

https://www.ibm.com/able/guidelines/ci162/keyboard.html

## Steps to reproduce the issue

Using only a keyboard

1. navigate within a modal that has overflowing content
2. notice you can never focus the content, making it so you can't scroll it unless you are using a mouse

> Please create a reduced test case in CodeSandbox

Includes the HTML fix:
https://codesandbox.io/embed/modal-overflow-content-srskh

## Additional information

* Need to add `tabindex=0` to the content container so that it is focusable.  This should be done only when the content overflows for the best experiences (dynamically via JS).
* Need input from @carbon-design-system/design around what the focus style for the content container should be

### PR DESCRIPTION
**Pull request by elizabethsjudd**
Closes #3196 

Modals that have content need to be focusable to allow keyboard users the ability to scroll the content.  Added a demo to show how developers can change the HTML to allow this and updated Modal styles to use the designs from https://github.com/carbon-design-system/carbon/issues/3261#issuecomment-507864039

#### Changelog

**New**

- Modal demo with scrolling content
- Modal content container focus styles

#### Testing / Reviewing

Test the keyboard users can scroll the content within modals and visual styles of focus state

### CODE REVIEW

File: packages/components/src/components/modal/_modal.scss [hunk 1]
```diff
@@ -135,6 +135,14 @@
     margin-bottom: $carbon--spacing-09;
     color: $text-01;
     font-weight: 400;
+
+    &:focus {
+      // @question: couldn't find a helper that this the wider focus style
```
Comment (emyarod) [inline packages/components/src/components/modal/_modal.scss]: I believe the `focus-outline('outline')` mixin can be used here

Comment (elizabethsjudd) [reply packages/components/src/components/modal/_modal.scss]: That did the trick.  Thanks @emyarod!

**Review by shixiedesign — APPROVED** [review]
From what I understand, this PR fixes the scrolling in modal. Looks like it worked! 🎉 Below is what I see: 
Aug-06-2019 10-54-17

**Review by asudoh — APPROVED** [review]
LGTM 👍 - Thanks @elizabethsjudd!

**Pull request merged**